File: NexusFinance.API/Models/Entities/Presupuesto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace NexusFinance.API.Models.Entities;

[Index("UsuarioId", "EstaActivo", "PeriodoPresupuesto", Name = "IX_Presupuestos_Usuario_Activo")]
public partial class Presupuesto
{
    [Key]
    public Guid PresupuestoId { get; set; }

    public Guid UsuarioId { get; set; }

    [StringLength(100)]
    public string NombrePresupuesto { get; set; } = null!;

    [StringLength(20)]
    public string PeriodoPresupuesto { get; set; } = null!;

    public DateTime FechaInicio { get; set; }

    public DateTime? FechaFin { get; set; }

    [Column(TypeName = "decimal(18, 2)")]
    public decimal PresupuestoTotal { get; set; }

    public bool? EstaActivo { get; set; }

    public DateTime? FechaCreacion { get; set; }

    public DateTime? FechaActualizacion { get; set; }

    [InverseProperty("Presupuesto")]
    public virtual ICollection<CategoriasPresupuesto> CategoriasPresupuestos { get; set; } = new List<CategoriasPresupuesto>();

    [ForeignKey("UsuarioId")]
    [InverseProperty("Presupuestos")]
    public virtual Usuario Usuario { get; set; } = null!;
}

File: NexusFinance.WEB/src/types/presupuesto.ts
export interface Presupuesto {
  presupuestoId: string;
  usuarioId: string;
  nombrePresupuesto: string;
  periodoPresupuesto: string;
  fechaInicio: string; // DateOnly se convierte a string
  fechaFin?: string; // DateOnly opcional
  presupuestoTotal: number;
  estaActivo?: boolean;
  fechaCreacion?: string;
  fechaActualizacion?: string;
}

export interface CrearPresupuestoDTO {
  usuarioId: string;
  nombrePresupuesto: string;
  periodoPresupuesto: string;
  fechaInicio: string;
  fechaFin?: string;
  presupuestoTotal: number;
}

export interface ActualizarPresupuestoDTO {
  presupuestoId: string;
  nombrePresupuesto?: string;
  periodoPresupuesto?: string;
  fechaInicio?: string;
  fechaFin?: string;
  presupuestoTotal?: number;
  estaActivo?: boolean;
}

export interface PresupuestoFormData {
  nombrePresupuesto: string;
  periodoPresupuesto: string;
  fechaInicio: string;
  fechaFin?: string;
  presupuestoTotal: number;
}

export interface PresupuestoConCategorias extends Presupuesto {
  categorias?: CategoriaPresupuesto[];
}

export interface CategoriaPresupuesto {
  categoriaPresupuestoId: string;
  presupuestoId: string;
  categoriaId: string;
  nombreCategoria: string;
  montoAsignado: number;
  montoGastado?: number;
  estaActivo?: boolean;
  fechaCreacion?: string;
  fechaActualizacion?: string;
} 